libsolidity/interface/StandardCompiler.cpp

```diff
@@ -409,12 +409,17 @@ boost::optional<Json::Value> StandardCompiler::parseOptimizerSettings(Json::Valu
 			return *error;
 		if (auto error = checkOptimizerDetail(details, "yul", settings.runYulOptimiser))
 			return *error;
+		if (settings.runYulOptimiser)
+			settings.optimizeStackAllocation = true;
 		if (details.isMember("yulDetails"))
 		{
-			if (!_jsonInput["yulDetails"].isObject())
-				return formatFatalError("JSONError", "The \"yulDetails\" optimizer setting has to be a JSON object.");
-			if (!_jsonInput["yulDetails"].getMemberNames().empty())
-				return formatFatalError("JSONError", "The \"yulDetails\" optimizer setting cannot have any settings yet.");
+			if (!details.isMember("yul"))
+				return formatFatalError("JSONError", "\"Providing yulDetails requires Yul optimizer to be enabled.");
```

Comment:
But the condition will also be true if the yul member is set to false, won't it? Shouldn't we either require the *value* to be in fact true or remove the check entirely?

Reply:
if yul is set to false, then settings.optimizeStackAllocation should be force to false, too...

Reply:
My point is that
```
  yulDetails: {}
```
will result in the error "Providing yulDetails requires Yul optimizer to be enabled."
Whereas:
```
  yul: false,
  yulDetails: {}
```
Will not error out, which is inconsistent.

Or am I missing something?

Reply:
Ah, now I understand! So this should be changed to `if (!details.isMember("yul") || details["yul"].asBool() != true)`?

Reply:
Yep.

Reply:
Or maybe better and simpler: ``if (!settings.runYulOptimiser)``.